Source/WebCore/platform/mediastream/ios/AVAudioSessionCaptureDeviceManager.mm

```diff
@@ -277,15 +278,25 @@ - (void)routeDidChange:(NSNotification *)notification
 Vector<AVAudioSessionCaptureDevice> AVAudioSessionCaptureDeviceManager::retrieveAudioSessionCaptureDevices() const
 {
     auto defaultMicrophoneInformation = computeDefaultMicrophoneInformation();
-    if (!defaultMicrophoneInformation && !m_lastDefaultMicrophone)
-        m_lastDefaultMicrophone = [m_audioSession currentRoute].inputs.firstObject;
+    auto currentInput = [m_audioSession currentRoute].inputs.firstObject;
+    if (currentInput) {
+        if (currentInput != m_lastDefaultMicrophone.get()) {
+            auto device = AVAudioSessionCaptureDevice::create(currentInput, currentInput);
+            callOnWebThreadOrDispatchAsyncOnMainThread(makeBlockPtr([device = crossThreadCopy(WTFMove(device))] () mutable {
+#if ENABLE(MEDIA_STREAM)
+                CoreAudioSharedUnit::singleton().handleNewCurrentMicrophoneDevice(WTFMove(device));
+#endif
```

Comment:
Isn't this entire file `ENABLE_MEDIA_STREAM` only?

Reply:
Done